rules/opentitan_test.bzl

```diff
@@ -313,20 +313,33 @@ def opentitan_functest(
             # Copy `cw310` for each `target`. This is not a deep copy, thus we
             # also copy tags below.
             cw310_ = cw310_params() if not cw310 else dict(cw310.items())
-            if "test_rom" in target:
+            cw310_["tags"] = [t for t in cw310_["tags"]]
+
+            # If the cw310 parameter was not provided or was provided without
+            # the bitstream field, determine the bitstream argument based on
+            # the target. Otherwise, use the provided bitstream argument.
+            DEFAULT_BITSTREAM = {
+                "cw310_test_rom": "@//hw/bitstream:test_rom",
+                "cw310_rom": "@//hw/bitstream:rom",
+            }
+            if (cw310 == None) or (cw310.get("bitstream") == None):
+                cw310_["bitstream"] = DEFAULT_BITSTREAM[target]
+
+            # If the bitstream field was provided, it will already have been copied into cw310_
```

Comment:
Nit: It might be clearer if you inline this logic back into the if-else branches. In the "cw310_test_rom" branch, you could just say `cw310_.setdefault("bitstream", "@//hw/bitstream:test_rom")`.